crates/bevy_asset/src/asset_server.rs

```diff
@@ -216,6 +216,47 @@ impl AssetServer {
         self.load_untyped(path).typed()
     }
 
+    /// Create a new asset from another one
+    pub fn create_from<FROM: Asset, TO: Asset, F>(
+        &self,
+        from_handle: Handle<FROM>,
+        transform: F,
+    ) -> Handle<TO>
+    where
+        F: FnOnce(&FROM) -> Option<TO>,
+        F: Send + 'static,
+    {
+        let new_handle = HandleId::random::<TO>();
+        let to_handle = Handle::strong(
+            new_handle,
+            self.server.asset_ref_counter.channel.sender.clone(),
+        );
+
+        let asset_lifecycles = self.server.asset_lifecycles.read();
+        if let Some(asset_lifecycle) = asset_lifecycles.get(&FROM::TYPE_UUID) {
+            asset_lifecycle.create_asset_from(
+                from_handle.into(),
+                new_handle,
+                TO::TYPE_UUID,
+                Box::new(|asset: &dyn AssetDynamic| {
+                    if let Some(transformed) = transform(
+                        asset
+                            .downcast_ref::<FROM>()
+                            // this downcast can't fail as we know the actual types here
+                            .expect("Error converting an asset to it's type, please open an issue in Bevy GitHub repository"),
```

Comment:
```suggestion
                            .expect("Error converting an asset to its type, please open an issue in Bevy GitHub repository"),
```

Reply:
Is there no end-user workaround for this?

Reply:
it could be logged and ignored instead of `expect`, but this one actually can't happen as it's the one place where we know the actual types and they are constrained by the function signature